### PR TITLE
Add new os_security_groups_facts module

### PR DESCRIPTION
**Pull request by Durburz**
##### SUMMARY
Add module to fetch security group data from OpenStack. 
To manage a secure OpenStack environment, it is neccessary to get information about existing security groups and rules.

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

cloud/openstack/os_security_group_facts

### CODE REVIEW

**Comment by Durburz** [comment]
bot_status

**Comment by Durburz** [comment]
-label affects_2.4
+label affects_2.5

**Comment by Durburz** [comment]
-affects_2.4

**Comment by Durburz** [comment]
-label affects_2.4

**Comment by Durburz** [comment]
+label affects_2.5
-label affects_2.4

**Comment by Durburz** [comment]
-label affects_2.4

**Comment by Durburz** [comment]
bot_status

**Comment by Durburz** [comment]
shipit

File: lib/ansible/modules/cloud/openstack/os_security_group_facts.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+try:
+    import shade
+    HAS_SHADE = True
+except ImportError:
+    HAS_SHADE = False
+
+
+def main():
+    argument_spec = openstack_full_argument_spec()
```
Comment (shreddedbacon) [inline lib/ansible/modules/cloud/openstack/os_security_group_facts.py:117]: Allow filters
```
argument_spec = openstack_full_argument_spec(filters=dict(required=False, type='dict', default=None))
```

File: lib/ansible/modules/cloud/openstack/os_security_group_facts.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    module = AnsibleModule(argument_spec)
+
+    if not HAS_SHADE:
+        module.fail_json(msg='shade is required for this module')
+
+    try:
+        cloud = shade.openstack_cloud(**module.params)
+        security_groups = cleanProtocol(cloud.list_security_groups())
```
Comment (shreddedbacon) [inline lib/ansible/modules/cloud/openstack/os_security_group_facts.py:125]: Allow filters
```
security_groups = cleanProtocol(cloud.list_security_groups(module.params['filters']))
```

**Review by shreddedbacon — COMMENTED** [review]
Being able to apply filters would be great

**Comment by mnaser** [comment]
needs_revision

**Comment by sshnaidm** [comment]
Thanks for submitting patch for Openstack Ansible modules!
We moved Openstack Ansible modules to Openstack repositories.
Next patches should be submitted not with Ansible Github but with
Openstack Gerrit: https://review.opendev.org/#/q/project:openstack/ansible-collections-openstack
Please submit your code there from now.
Thanks for your contribution and sorry for inconvienience.

**Pull request closed without merging**